Documentation/AdditionalFunctionality/PushNotifications.md

```diff
@@ -0,0 +1,133 @@
+# Push Notifications
+
+The `NotificationCenter` handles the registration and sending of **push notifications** to Apple Push Notification Service (APNS) and Firebase Cloud Messaging (FCM). APNS can send out push notifications to iOS, macOS and tvOS devices, while FCM handles messaging for iOS, Android and Web Push.
+
+## Configuration
+
+In order to send push notifications, the Apodini web server needs to be authenticated to at least one push notification service. These services are added to a web server by defining them in the `configuration` computed property.
+
+This example adds APNS to the web server.
+
+```swift
+var configuration: some Configuration {
+    APNSConfiguration(.pem(pemPath: "certificate.pem"), identifier: "de.tum.in.ase.Example", environment: .sandbox)
+}
+```
+
+## Usage
+
+The `NotificationCenter` can be injected to any `Component` by using the `@Environment` property wrapper.
+
+```swift
+struct NewsAlertComponent: Component {
+    @Environment(\.notificationCenter) var notificationCenter: NotificationCener
+
+    // ...
+}
+```
+
+## Device Registration
+
+Devices can be registered to the `NotificationCenter` by conforming to the `Device` protocol. This protocol consists of the following proerties:
+
+- **type**: This property specifies which push notification service to use.
+- **token**: The device id for one app used by the push notification service
+- **subscriptions**: Subscriptions are used to group `Device`s together. Example: All devices that want to receive news alerts for a specific topic.
+
+```swift
+public struct Device: Content {
```

Comment:
What is `Content`?

Reply:
`Content` is Vapor specific for decoding and encoding. I removed it from the proposal.